File: web/src/shared/types/console.ts
// 控制台公共类型只描述跨页面查询结果，业务写操作仍留在各功能切片中。
import type { ActionRun, Incident, RemediationPlan } from "../../features/incidents/types";

export interface ConsoleOverview {
  incident_count: number;
  active_incident_count: number;
  resolved_incident_count: number;
  execution_count: number;
  successful_execution_count: number;
  asset_count: number;
  plugin_count: number;
  access_mode: string;
}

export interface ManagedAsset {
  id: string;
  name: string;
  kind: string;
  node_id: string;
  status: "healthy" | "degraded" | string;
  address: string;
  responsibilities: string[];
}

export interface TopologyEdge {
  source_asset_id: string;
  target_asset_id: string;
  relation: string;
}

export interface TopologySnapshot {
  nodes: ManagedAsset[];
  edges: TopologyEdge[];
}

export interface IncidentContext {
  incident: Incident;
  plan: RemediationPlan | null;
  executions: ActionRun[];
}

export interface Capability {
  id: string;
  name: string;
  category: string;
  state: "online" | "degraded" | "experiment" | "disabled" | string;
  description: string;
}

export interface SafeSettings {
  access_mode: string;
  admin_actions_enabled: boolean;
  trust_proxy_client_ip: boolean;
  prometheus_enabled: boolean;
  loki_enabled: boolean;
  llm_enabled: boolean;
  plugin_runtime_enabled: boolean;
  real_actions_mode: string;
  agent_grpc_enabled: boolean;
  secrets_exposed: boolean;
}

export interface DiagnosticStatus {
  provider: string;
  model: string;
  rag: string;
  degraded: boolean;
}

export interface DiagnosticAnswer {
  diagnosis: {
    summary: string;
    root_cause: string;
    recommended_action: string;
    confidence: number;
  };
  citations: string[];
  degraded: boolean;
}

export interface PluginManifest {
  plugin_id: string;
  name: string;
  version: string;
  api_version: string;
  capabilities: string[];
  permissions: string[];
}

export interface ChangeEvent {
  id: string;
  provider: string;
  event_type: string;
  service: string;
  revision: string;
  status: string;
  occurred_at: string;
}

File: control-plane/src/aiops_control/domain/console.py
"""控制台只读视图模型。

输入：领域仓储中的 Incident、修复计划、执行记录以及部署时声明的组件能力。
处理：只描述控制台需要展示的稳定字段，不包含数据库或 HTTP 细节。
输出：资产状态、系统能力和汇总指标等不可变查询结果。
副作用：无；所有状态推导均由应用层完成。
"""

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class ManagedAssetView:
    """控制台展示的资产及其当前健康摘要。"""

    id: str
    name: str
    kind: str
    node_id: str
    status: str
    address: str
    responsibilities: tuple[str, ...]


@dataclass(frozen=True, slots=True)
class CapabilityView:
    """一个可学习、可观察但不夸大完成度的系统能力。"""

    id: str
    name: str
    category: str
    state: str
    description: str


@dataclass(frozen=True, slots=True)
class ConsoleOverview:
    """首页使用的轻量聚合统计。"""

    incident_count: int
    active_incident_count: int
    resolved_incident_count: int
    execution_count: int
    successful_execution_count: int
    asset_count: int
    plugin_count: int
    access_mode: str
